lib/merchants_repository.rb

```diff
@@ -1,7 +1,45 @@
 class MerchantsRepository
-  attr_reader :merchant_array
+  attr_reader :all
 
   def initialize(merchant_array)
-    @merchant_array = merchant_array
+    @all = merchant_array
+  end
+
+  def find_by_id(id)
+    @all.find { |merchant| merchant.id == id }
+  end
+
+  def find_by_name(name)
+    @all.find { |merchant| merchant.name.upcase == name.upcase }
+  end
+
+  def find_all_by_name(name)
+    @all.find_all { |merchant| merchant.name.upcase.include? name.upcase }
+  end
+
+  def create(attributes)
+    max_id = 0
+    @all.each do |merchant|
+      max_id = merchant.id if merchant.id > max_id
+    end
```

Comment:
You can definitely condense this down with a .max enum. I'd look up the spaceship operator to help with that - it's really helpful for .max enums!